Application/components/FriendsColumn/RemoveFriendModal.tsx

```diff
@@ -0,0 +1,148 @@
+import React, { useState } from 'react';
+import { Modal, Text, Stack, Button, useMantineTheme, MultiSelect, Tooltip, Paper } from '@mantine/core';
+import { IconUserMinus } from '@tabler/icons-react';
+import { useUser } from '@clerk/nextjs';
+import { useCache } from '@/contexts/queryCacheContext';
+import { useFriendList } from '@/hooks/useFriendList';
+import { showNotification } from '@mantine/notifications';
+
+interface Friend {
+  id: string;
+  username: string;
+}
+
+interface RemoveFriendResponse {
+  message: string;
+}
+
+export function RemoveFriendModal(): JSX.Element {
+  const { user } = useUser();
+  const theme = useMantineTheme();
+  const [opened, setOpened] = useState<boolean>(false);
+  const { lastFetched, setLastFetched } = useCache();
+  const [selectedFriend, setSelectedFriend] = useState<string[]>([]);
+  const [errorMessages, setErrorMessages] = useState<{ friends: string }>({ friends: '' });
+  const { list: friendsList } = useFriendList({ lastFetched, setLastFetched });
+
+  const friendOptions = friendsList.map((friend: Friend) => ({
+    value: friend.id,
+    label: friend.username,
+  }));
+
+  const handleOpenModalClick = () => {
+    if (friendsList.length > 0) {
+      setOpened(true);
+    } else {
+      showNotification({
+        title: 'No Friends to Remove',
+        message: 'You currently have no friends to remove.😢',
+        color: 'red',
+      });
+    }
+  };
+
+  const handleRemoveFriendClick = async (): Promise<void> => {
+    if (selectedFriend.length === 0) {
+      setErrorMessages({ friends: "Please select a friend to remove." });
+      return;
+    }
+
+    try {
+      const response = await fetch('/api/Remove-friend', {
+        method: 'POST',
+        headers: {
+          'Content-Type': 'application/json',
+        },
+        body: JSON.stringify({
+          userID: user?.id,
```

Comment:
It would be preferrable if you used the `useEffect` pattern we have been using across all files to keep the user ID in a state and ensure that it exists before doing this. This can be risky because there is a short micro delay between the time our `useUser()` hook is instantiated and the time it returns actual user data.